File: samples/client/3_0_3_unit_test/java/src/main/java/org/openapijsonschematools/client/schemas/validation/ValidationData.java
package org.openapijsonschematools.client.schemas.validation;

import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.List;

public record ValidationData(
    JsonSchema<?> schema,
    @Nullable Object arg,
    ValidationMetadata validationMetadata,
    @Nullable List<PathToSchemasMap> containsPathToSchemas,
    @Nullable PathToSchemasMap patternPropertiesPathToSchemas,
    @Nullable PathToSchemasMap ifPathToSchemas,
    @Nullable PathToSchemasMap knownPathToSchemas
) {
    public ValidationData(
            JsonSchema<?> schema,
            @Nullable Object arg,
            ValidationMetadata validationMetadata
    ) {
        this(schema, arg, validationMetadata, null, null, null, null);
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/ValidationData.kt
package org.openapijsonschematools.client.schemas.validation

data class ValidationData(
    val schema: JsonSchema<*>,
    val arg: Any?,
    val validationMetadata: ValidationMetadata,
    val containsPathToSchemas: List<PathToSchemasMap>? = null,
    val patternPropertiesPathToSchemas: PathToSchemasMap? = null,
    val ifPathToSchemas: PathToSchemasMap? = null,
    val knownPathToSchemas: PathToSchemasMap? = null
)